addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2391,8 +2391,8 @@ var SnippetsMenu = Widget.extend({
                 'base_target': target,
                 'selector': self._computeSelectorFunctions(selector, exclude, target, noCheck),
                 '$el': $style,
-                'drop-near': $style.data('drop-near') && self._computeSelectorFunctions($style.data('drop-near'), '', false, noCheck, true, excludeParent),
-                'drop-in': $style.data('drop-in') && self._computeSelectorFunctions($style.data('drop-in'), '', false, noCheck),
+                'drop-near': $style.data('drop-near') && self._computeSelectorFunctions($style.data('drop-near'), '.o_editable_media', false, noCheck, true, excludeParent),
+                'drop-in': $style.data('drop-in') && self._computeSelectorFunctions($style.data('drop-in'), '.o_editable_media', false, noCheck),
```

Comment:
My fix is a bit hacky... the problem, in my opinion, is that the same filter function is used to determine whether an option should be started and whether a block can be dropped at a certain point. 
```js
let filterFunc = function () {
    // Exclude what it is asked to exclude.
    if ($(this).is(exclude)) {
        return false;
    }
    // `o_editable_media` bypasses the `o_not_editable` class.
    if (this.classList.contains('o_editable_media')) {
        return shouldEditableMediaBeEditable(this);
    }
    return !$(this).is('.o_not_editable :not([contenteditable="true"])');
};
```
The `o_editable_media` class should only influence behavior when determining whether an option can be started IMO. Since in stable you can't change the function signature, I'm proposing this change, which is not elegant I admit (you can never drop next to an `o_editable_media`). This will be OK for team and social media blocks. (but maybe not for a custom block made by a user :grimacing: )

In master I'd be in favor of having the "context" (drop or option ?) in the filter function to take into account (or not) the `o_editable_media` class. I'm curious to know what you think @qsm-odoo ?
In master something like that:
```js
_computeSelectorFunctions: function (selector, exclude, target, noCheck, isChildren, excludeParent, forDrop) {
...
let filterFunc = function () {
    ...
    if (!forDrop && this.classList.contains('o_editable_media')) {
        return shouldEditableMediaBeEditable(this);
    }
    return !$(this).is('.o_not_editable :not([contenteditable="true"])');
};
...
_computeSnippetTemplates: function (html) {
...
var option = {
   ...
    'selector': self._computeSelectorFunctions(selector, exclude, target, noCheck, false, false, false, false),
    '$el': $style,
    'drop-near': $style.data('drop-near') && self._computeSelectorFunctions($style.data('drop-near'), '', false, noCheck, true, excludeParent, true),
    'drop-in': $style.data('drop-in') && self._computeSelectorFunctions($style.data('drop-in'), '', false, noCheck, false, false, false, true),
```

Reply:
@Guillaume-gdi Agreed, now that I re-see this, I wonder if the `noCheck` parameter is not broken as well: we check for o_not_editable even if `noCheck` is true... we should not. If `noCheck`, we should just consider the given `exclude`. Maybe fix that at the same time (maybe same PR, different commit).

For the o_editable_media thing, I guess we could make a better hack in stable and already simulate your `forDrop` param of master. At the start of the method:
```javascript
exclude = exclude || '';
const forDropID = 'FOR_DROP';
const forDrop = exclude.startsWith(forDropID);
exclude = exclude.substring(forDropID.length);
```
(untested of course). Then you can give `"FOR_DROP"` as parameter instead of your `.o_editable_media` as exclude which as you explained could break custom cases.